CHANGELOG.md

```diff
@@ -5,8 +5,12 @@ All notable changes to this project will be documented in this file.
 
 ### Added
 
+-   ##1576 : Add test functions which pass and return classes.
+
 ### Fixed
 
+-   ##1576 : Fix Destructor Call for C and Fortran.
```

Comment:
What is the problem that you are fixing?

Reply:
Fixed https://github.com/pyccel/pyccel/pull/1576/commits/b6f1c7054ac8904ecf769c09923be7befd883a19.

Reply:
I still don't know what this means :sweat_smile: Can you explain in more words here

Reply:
Here as in on the thread, not in the changelog

Reply:
Before the automatique call of the Distructor was not perfect, sometimes it was called even if the class is still used.
Now I fixed that problem and it's called at the right place.

Reply:
Do you know what conditions caused this?

Reply:
it's in here : https://github.com/pyccel/pyccel/blob/c163e184f5d393e8fb5289b35159e4a8c413dced/pyccel/parser/semantic.py#L3826

Reply:
I mean what user code causes the destructor to be called in the wrong place